src/ManifoldMuseum.jl

```diff
@@ -3,10 +3,24 @@ module ManifoldMuseum
 import Base: isapprox, exp, log
 import LinearAlgebra: dot, norm
 
-
 abstract type Manifold end
 
-isapprox(m::Manifold, x, y; kwargs...) = isapprox(x, y; kwargs...)
+"""
+   IsDecoratorManifold
+
+A `Trait` to mark a manifold as a decorator type. For any function that is only
+implemented for a decorator (i.e. a Manifold with `@traitimpl
+IsDecoratorManifold{M}`), a specific function should be implemented as a
+`@traitfn`, that transparently passes down through decorators, i.e.
+
+```
+@traitfn myFeature(M::Mt, k...) where {Mt; IsNice{Mt}} = myFeature(M.manifold, k...)
+myFeature(M::MyManifold, k...) = #... my explicit impementation
```

Comment:
`IsNice{Mt}` should be `IsDecoratorManifold{Mt}`. Should also document that the pass through can be overriden if desired. e.g.
`myFeature(MyDecoratorManifold{MyManifold}, k...) =  #... my explicit override

Reply:
We could also document the easier syntax `@traitfn myFeature(M::::IsDecoratorManifold, k...) = myFeature(M.manifold, k...)`